src/codemods/__testfixtures__/colors-to-css-vars/color-as-type.output.tsx

```diff
@@ -0,0 +1,7 @@
+import { ReadBareCssColorVariable } from '@freenow/wave';
+
+export interface LabelVariant {
+    color?: ReadBareCssColorVariable
```

Comment:
```suggestion
    color?: ReadCssColorVariable | (string & {})
```

Reply:
Do you think there would be any issue using just `ReadCssColorVariable | string`? I've tested it and it was fine, but maybe I've overlooked something.

The issue with `{}` is that there is an Eslint rule which bans `{}` as a type because it means "any non-nullish value". I'm thinking that if a project has this rules enabled they will need manual changes here 🤔

Reply:
When using `ReadCssColorVariable | string` we would lose the intellisense, I've tried it with `ReadSemanticCssColorVariable | (string & NonNullable<unknown>)` instead and we keep the intellisense while avoiding the ESLint rule. `NonNullable<unknown>` works because internally it's just doing `unknown & {}`, as long as we keep the `{}` the intellisense works. Wdyt?

Reply:
I propose to disable the rule for this line (with the reasoning) instead of hacks

Reply:
Even though I like the idea adding comments is not straightforward, different types of nodes need different types of comments, for example if I just add a comment to the existing `TSTypeReference` node the comment will be written before the actual type 😅, instead I need to find the parent node and then attach the comment there, but the parent node can have different types, and attaching a trailing comment works differently for each type.

Just to give an example of the complexity of adding comments I've found [this codemod](https://github.com/amanda-mitchell/suppress-eslint-errors/blob/main/transforms/suppress-eslint-errors.js), which disables eslint rules and adds a comment on top, ours would be a bit simpler, but not by a lot. 

I've also tried adding an `eslint-disable-next-line` comment on the line before the type usage but it also didn't seem straightforward.

Nevertheless I'll take a deeper look next week to see if I can come up with an easy solution that works for all our use cases.

Reply:
In the end I've been able to add a leading eslint-disable with the reasoning comment each time we replace a usage of `Colors` as a type 👌🏽